dockerfiles/docker-entrypoint.sh

```diff
@@ -96,50 +103,51 @@ update_conf() {
     PATCH_REF_CONF="$4"
 
     if [ -f "${CONF}" ]; then
-	if [ "${REF_CONF}" -nt "${CONF}" ]; then
-	    # There is a new version
-	    if [ $FORCE_CONF_UPDATE -ne 0 ]; then
-		# Replace the current configuration
-		printf '⚠️  Automaticaly update %s to the new version\n' "${CONF}"
-		if [ ! -f "${OLD_CONF}" ]; then
-		    printf 'The previous configuration is saved to %s\n' "${OLD_CONF}"
-		    mv "${CONF}" "${OLD_CONF}"
-		fi
-		cp "${REF_CONF}" "${CONF}"
-		$PATCH_REF_CONF "${CONF}"
-	    else
-		# Keep the current configuration
-		printf '⚠️  Check new version %s to make sure searx is working properly\n' "${NEW_CONF}"
-		cp "${REF_CONF}" "${NEW_CONF}"
-		$PATCH_REF_CONF "${NEW_CONF}"
-	    fi
-	else
-	    printf 'Use existing %s\n' "${CONF}"
-	fi
+        if [ "${REF_CONF}" -nt "${CONF}" ]; then
+            # There is a new version
+            if [ "$FORCE_CONF_UPDATE" -ne 0 ]; then
+                # Replace the current configuration
+                printf '⚠️  Automaticaly update %s to the new version\n' "${CONF}"
+                if [ ! -f "${OLD_CONF}" ]; then
+                    printf 'The previous configuration is saved to %s\n' "${OLD_CONF}"
+                    mv "${CONF}" "${OLD_CONF}"
+                fi
+                cp "${REF_CONF}" "${CONF}"
+                $PATCH_REF_CONF "${CONF}"
+            else
+                # Keep the current configuration
+                printf '⚠️  Check new version %s to make sure searx is working properly\n' "${NEW_CONF}"
+                cp "${REF_CONF}" "${NEW_CONF}"
+                $PATCH_REF_CONF "${NEW_CONF}"
+            fi
+        else
+            printf 'Use existing %s\n' "${CONF}"
+        fi
     else
-	printf 'Create %s\n' "${CONF}"
-	cp "${REF_CONF}" "${CONF}"
-	$PATCH_REF_CONF "${CONF}"
+        printf 'Create %s\n' "${CONF}"
+        cp "${REF_CONF}" "${CONF}"
+        $PATCH_REF_CONF "${CONF}"
     fi
 }
 
 # make sure there are uwsgi settings
-update_conf ${FORCE_CONF_UPDATE} "${UWSGI_SETTINGS_PATH}" "/usr/local/searx/dockerfiles/uwsgi.ini" "patch_uwsgi_settings"
+update_conf "${FORCE_CONF_UPDATE}" "${UWSGI_SETTINGS_PATH}" \
+	    "/usr/local/searx/dockerfiles/uwsgi.ini" "patch_uwsgi_settings"
 
 # make sure there are searx settings
-update_conf "${FORCE_CONF_UPDATE}" "${SEARX_SETTINGS_PATH}" "/usr/local/searx/searx/settings.yml" "patch_searx_settings"
+update_conf "${FORCE_CONF_UPDATE}" "${SEARX_SETTINGS_PATH}" \
+	    "/usr/local/searx/searx/settings.yml" "patch_searx_settings"
 
 # dry run (to update configuration files, then inspect them)
 if [ $DRY_RUN -eq 1 ]; then
     printf 'Dry run\n'
     exit
 fi
 
-#
 touch /var/run/uwsgi-logrotate
 chown -R searx:searx /var/log/uwsgi /var/run/uwsgi-logrotate
 unset MORTY_KEY
 
 # Start uwsgi
-printf 'Listen on %s\n' "${BIND_ADDRESS}"
+echo 'Listen on %s' "${BIND_ADDRESS}"
```

Comment:
it should be `echo 'Listen on' "${BIND_ADDRESS}"` without `%s` ?

Reply:
> or keep printf as above (line 124, 127, ....) ?

I tend to used echo because I think it is more readable, but this is my very personal opinion. More experienced users say it is better to use printf ([Basically, it's a portability (and reliability) issue](https://unix.stackexchange.com/questions/65803/why-is-printf-better-than-echo/65819#65819)).

on second thought, I would say it is better to stay with printf.